src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_RANDOM_NUMBER = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(FigureType.values().length);
+        switch (FigureType.values()[index]) {
+            case CIRCLE:
+                return new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANDOM_NUMBER));
+            case SQUARE:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANDOM_NUMBER));
+            case RECTANGLE:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANDOM_NUMBER),
+                        random.nextInt(MAX_RANDOM_NUMBER));
+            case RIGHT_TRIANGLE:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANDOM_NUMBER));
+            default:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANDOM_NUMBER),
+                        random.nextInt(MAX_RANDOM_NUMBER),
+                        random.nextInt(MAX_RANDOM_NUMBER));
```

Comment:
you create array with 5 figures just to pick one of them. Lets use switch-case construction. We can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability